api/HTMLCanvasElement.json

```diff
@@ -195,9 +195,58 @@
             }
           }
         },
-        "desynchronized": {
+        "options_alpha_parameter": {
           "__compat": {
-            "description": "<code>desynchronized</code> attribute",
+            "description": "<code>options.alpha</code> parameter",
+            "spec_url": "https://www.khronos.org/registry/webgl/specs/latest/1.0/#WEBGLCONTEXTATTRIBUTES",
```

Comment:
I would say https://www.khronos.org/registry/webgl/specs/latest/1.0/#5.2.1 is a better URL for this, or links to the individual bits of that section if IDs exist.

Note also that this and other attributes may be supported by 2d canvas and webgl2, in which case those should also be linked.

Reply:
`alpha` is also supported by the "2d" and "bitmaprenderer" contexts:
https://html.spec.whatwg.org/multipage/canvas.html#2d-context-creation-algorithm
https://html.spec.whatwg.org/multipage/canvas.html#imagebitmaprenderingcontext-creation-algorithm

Looking at this I thought we should group these parameters under the context features. In other words, wthat e should have these entries:
- api.HTMLCanvasElement.getContext.2d_context.options_alpha_parameter
- api.HTMLCanvasElement.getContext.2d_context.options_desynchronized_parameter
- api.HTMLCanvasElement.getContext.2d_context.options_colorSpace_parameter
- api.HTMLCanvasElement.getContext.2d_context.options_willReadFrequently_parameter
- api.HTMLCanvasElement.getContext.bitmaprenderer_context.options_alpha_parameter
- api.HTMLCanvasElement.getContext.webgl_context.options_alpha_parameter
- api.HTMLCanvasElement.getContext.webgl_context.options_antialias_parameter
- api.HTMLCanvasElement.getContext.webgl_context.options_depth_parameter
- api.HTMLCanvasElement.getContext.webgl_context.options_desynchronized_parameter
- api.HTMLCanvasElement.getContext.webgl_context.options_failIfMajorPerformanceCaveat_parameter
- api.HTMLCanvasElement.getContext.webgl_context.options_powerPreference_parameter
- api.HTMLCanvasElement.getContext.webgl_context.options_premultipliedAlpha_parameter
- api.HTMLCanvasElement.getContext.webgl_context.options_preserveDrawingBuffer_parameter
- api.HTMLCanvasElement.getContext.webgl_context.options_stencil_parameter
- api.HTMLCanvasElement.getContext.webgl2_context._REPEAT_THE_ABOVE_

However, due mainly due to the duplication between "webgl" and "webgl2" entries, this seems unwieldy. The parameters supported in more than one context (treating WebGL as one) are:
- `alpha` in 2d, bitmaprenderer and webgl(2)
- `desynchronized` in 2d and webgl(2)

So we could probably do merged entries representing both and use notes or `partial_implementation` if there's any complication. The downside is that which parameters that work for a given context won't be clear, and people might expect "colorSpace" to work with WebGL.

I'd like to hear from @ddbeck and @Elchi3 how they'd deal with this situation.

Reply:
> So we could probably do merged entries representing both and use notes or partial_implementation if there's any complication. The downside is that which parameters that work for a given context won't be clear, and people might expect "colorSpace" to work with WebGL.

As much as I hate all the length and repetition of the duplication, it has the benefit of being direct. Deduplicating seems to maximize complexity for readers and BCD contributors alike. And if we truly hate the longer option, then I imagine it'll be easier to collapse things that it would be to go the other direction.

That's my take. I don't feel strongly about this stuff, though.

Reply:
OK, I could definitely live with the deduplicated approach. If I squint and think about this like mixins merging together different things, I'm horrified by my suggestion to merge the entries :)

@vinyldarkscratch do you want to try rearranging the existing entries like that? No need to add what's missing I think, just sorting out what we have is more than enough for this PR :)

Reply:
Finished sorting out the options into their respective contexts!  I ended up having to fix some of the data as well, so I've re-titled this PR to increase its scope.  I added the missing `2d` context, but I haven't added any of the individual options -- that'll be a later thing!